sw/device/lib/dif/dif_pwrmgr.h

```diff
@@ -293,7 +303,7 @@ dif_pwrmgr_result_t dif_pwrmgr_init(dif_pwrmgr_params_t params,
  * @return The result of the operation.
  */
 DIF_WARN_UNUSED_RESULT
-dif_pwrmgr_result_t dif_pwrmgr_low_power_set_enabled(
+dif_pwrmgr_config_result_t dif_pwrmgr_low_power_set_enabled(
```

Comment:
This is a totally dumb question, but is it worth unifying all of these configuration bits into a single struct, seeing as it needs to be called every time we want to go into low-power?

Reply:
Not a dumb question at all, I also thought about the same thing but wanted to keep these separate for the sake of flexibility:
- Domain configuration can change at runtime, e.g. deep sleep or normal sleep, independent of wakeup and reset request configurations.
- Both wakeup and reset requests can be independently changed and locked. For some use-cases, it may be natural to have static configurations but for others these may need to be changed at runtime, e.g. disable wakeup due to something at some state but enable at some other state.
- Low power hint, on the other hand, is required before every transition to low power.